feat: add docs for repl and prompt modes (#3311)
heroku/cli: docs/prompt.md

# Heroku CLI Interactive Prompt

The interactive prompt helps you run Heroku CLI commands by guiding you through the required inputs. 

Prompting is particularly useful when you're unsure about what required or optional arguments and flags a command uses.

## Benefits

* Learn requirements for any command, like setting configuration vars

> * Run commands with multiple required flags, such as when creating new resources

Reviewer: ```suggestion
* Run commands with multiple required flags easily, such as when creating new resources
* ```

# Heroku CLI Interactive Prompt

> The Heroku CLI interactive prompt guides you through running commands by prompting for required arguments and flags.

Reviewer: ```suggestion
The interactive prompt helps you run Heroku CLI commands by guiding you through the required inputs. Prompting is particularly useful when you're unsure about what arguments and flags a command requires.
```

## Usage

```term
$ heroku --prompt COMMAND
```

> ## Description

Reviewer: ```suggestion
```

> ## Description

Reviewer: ```suggestion
```

> The interactive prompt helps you run Heroku CLI commands by guiding you through the required inputs. This is particularly useful when you're not sure about all the required arguments and flags for a command.

Reviewer: ```suggestion
```

> ### Features

Reviewer: ```suggestion
```

> * Prompts for required arguments

Reviewer: ```suggestion
```

* Prompts for required flags

> * Provides descriptions and examples for each input

Reviewer: ```suggestion
```

* Validates input before executing the command

### Example Session

```term
$ heroku --prompt config:set
Enter key: DATABASE_URL
Enter value: postgres://...
Setting DATABASE_URL and restarting example-app... done
```

> ## Common Use Cases

Reviewer: ```suggestion
```

* Setting configuration variables

> * Creating new resources

Reviewer: ```suggestion
```

> * Running commands with multiple required flags

Reviewer: ```suggestion
```

> * Learning command requirements

Reviewer: ```suggestion
```

> ## Related Topics

Reviewer: ```suggestion
## Additional Reading
```
